### PR TITLE
Search complex emojis with multiple equal skin tones by a single one

### PR DESCRIPTION
**Pull request by Envek**
Hello again! Here I want to propose one more usability enhancement for better compatibility with real world usage of emoji short codes.

**TL;DR**: Allows to find emoji 🧑🏽‍🤝‍🧑🏽 by shortened code `:people_holding_hands::skin-tone-3:` in addition to current `:people_holding_hands::skin-tone-3::skin-tone-3:` to improve compatibility.

**The problem**

Existing apps (e.g. Slack) and libraries (e.g. [EmojiMart](https://github.com/missive/emoji-mart) emoji picker) recognizes shorthand codes for **complex emojis that have multiple skin tone variation _axes_** for the case when _all_ of these axes has the same skin tone selected.

Example of such emoji is [People holding hands 🧑‍🤝‍🧑](https://emojipedia.org/people-holding-hands/) that can have dozens of skin tones variations where two people can have the same or different skin tones (e.g. 🧑🏽‍🤝‍🧑🏽 or 🧑🏻‍🤝‍🧑🏿).

To confuse things more, some other (older) emojis (like [Women holding hands 👭](https://emojipedia.org/women-holding-hands/) actually have special (separate) variants for the case when both people on the emoji has the same skin tone (also for historical reasons I believe). E.g. check codepoints for [Women Holding Hands: Light Skin Tone 👭🏻](https://emojipedia.org/women-holding-hands-light-skin-tone/) and [Women Holding Hands: Light Skin Tone, Dark Skin Tone 👩🏻‍🤝‍👩🏿](https://emojipedia.org/women-holding-hands-light-skin-tone-dark-skin-tone/).

You can check the difference yourself in the [emoji data set](https://raw.githubusercontent.com/iamcal/emoji-data/v14.0.0/emoji_pretty.json) (look for `skin_variations` field for `people_holding_hands` and `women_holding_hands` and you will see it)

In current emoji-datasource implementation, former light skin tone emoji can be found only by `:people_holding_hands::skin-tone-2::skin-tone-2:` while latter – only by `:two_women_holding_hands::skin-tone-2:`.

However, Slack correctly understands short codes for both `:people_holding_hands::skin-tone-2:` and `:two_women_holding_hands::skin-tone-2:`. EmojiMart also sends the same codes if skin tone is set (it doesn't support that multidimensional skin tone variatons at the time of writing) and they can't be recognized yet.

<details>
<summary>Full list of complex emoji with multiple skin tone dimensions and without short versions for the case when all skin tones are equal</summary>

```
man-heart-man
man-kiss-man
people_holding_hands
woman-heart-man
woman-kiss-man
woman-kiss-woman
```
</details>

**The solution**

Add `:people_holding_hands::skin-tone-2:` key to short code search hash map, pointing to the `:people_holding_hands::skin-tone-2::skin-tone-2:` emoji.

### CODE REVIEW

**Comment by jpalumickas** [comment]
Thanks @Envek ! Released v14.0.2 🎉
